src/porting.h

```diff
@@ -211,43 +211,58 @@ void initIrrlicht(irr::IrrlichtDevice * );
 	}
 
 #else // Posix
-
-	inline u32 getTimeS()
+	inline void _os_get_clock(struct timespec *ts)
 	{
+#if defined(__MACH__) && defined(__APPLE__)
+	// from http://stackoverflow.com/questions/5167269/clock-gettime-alternative-in-mac-os-x
+	// OS X does not have clock_gettime, use clock_get_time
+		clock_serv_t cclock;
+		mach_timespec_t mts;
+		host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
+		clock_get_time(cclock, &mts);
+		mach_port_deallocate(mach_task_self(), cclock);
+		ts.tv_sec = mts.tv_sec;
+		ts.tv_nsec = mts.tv_nsec;
+#else
+#ifdef CLOCK_MONOTONIC_RAW
```

Comment:
-----Original Message-----
From: "Auke Kok" <notifications@github.com>
Sent: ‎27/‎01/‎2016 16:52
To: "minetest/minetest" <minetest@noreply.github.com>
Subject: Re: [minetest] New timer design. (#3593)

In src/porting.h:
>  	{
> +#if defined(__MACH__) && defined(__APPLE__)
> +	// from http://stackoverflow.com/questions/5167269/clock-gettime-alternative-in-mac-os-x
> +	// OS X does not have clock_gettime, use clock_get_time
> +		clock_serv_t cclock;
> +		mach_timespec_t mts;
> +		host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
> +		clock_get_time(cclock, &mts);
> +		mach_port_deallocate(mach_task_self(), cclock);
> +		ts.tv_sec = mts.tv_sec;
> +		ts.tv_nsec = mts.tv_nsec;
> +#else
> +#ifdef CLOCK_MONOTONIC_RAW
sure, will do.
—
Reply to this email directly or view it on GitHub.